Exception in archiving of batch operations

### <!-- Component -->

<!-- Operate- -->

### <!-- Severity -->

<!-- Medium- -->

### <!-- Likelihood -->

<!-- High_ -->

### Description

Exception is logged in 8.8.0-SNAPSHOT:
```
java.util.concurrent.CompletionException: co.elastic.clients.util.MissingRequiredPropertyException: Missing required property 'TermsQuery.field'
<..>
```



### Steps to reproduce

1. Run C8 with CamundaExporter
2. Create some batch operations
3. Wait till 1 hour passes, so that finished batch operations are taken for archiving


### Current behavior

Exception is thrown:
```
java.util.concurrent.CompletionException: co.elastic.clients.util.MissingRequiredPropertyException: Missing required property 'TermsQuery.field'
Caused by: co.elastic.clients.util.MissingRequiredPropertyException: Missing required property 'TermsQuery.field'
	at co.elastic.clients.util.ApiTypeHelper.requireNonNull(ApiTypeHelper.java:76) ~[elasticsearch-java-8.16.4.jar:?]
	at co.elastic.clients.elasticsearch._types.query_dsl.TermsQuery.<init>(TermsQuery.java:78) ~[elasticsearch-java-8.16.4.jar:?]
	at co.elastic.clients.elasticsearch._types.query_dsl.TermsQuery.<init>(TermsQuery.java:64) ~[elasticsearch-java-8.16.4.jar:?]
	at co.elastic.clients.elasticsearch._types.query_dsl.TermsQuery$Builder.build(TermsQuery.java:189) ~[elasticsearch-java-8.16.4.jar:?]
	at io.camunda.exporter.tasks.archiver.ElasticsearchArchiverRepository.buildIdTermsQuery(ElasticsearchArchiverRepository.java:248) ~[camunda-exporter-8.8.0-SNAPSHOT.jar:8.8.0-SNAPSHOT]
	at io.camunda.exporter.tasks.archiver.ElasticsearchArchiverRepository.lambda$reindexDocuments$9(ElasticsearchArchiverRepository.java:159) ~[camunda-exporter-8.8.0-SNAPSHOT.jar:8.8.0-SNAPSHOT]
	at co.elastic.clients.elasticsearch.core.reindex.Source$Builder.query(Source.java:337) ~[elasticsearch-java-8.16.4.jar:?]
	at io.camunda.exporter.tasks.archiver.ElasticsearchArchiverRepository.reindexDocuments(ElasticsearchArchiverRepository.java:159) ~[camunda-exporter-8.8.0-SNAPSHOT.jar:8.8.0-SNAPSHOT]
	at io.camunda.exporter.tasks.archiver.ArchiverRepository.moveDocuments(ArchiverRepository.java:41) ~[camunda-exporter-8.8.0-SNAPSHOT.jar:8.8.0-SNAPSHOT]
	at io.camunda.exporter.tasks.archiver.BatchOperationArchiverJob.moveBatch(BatchOperationArchiverJob.java:61) ~[camunda-exporter-8.8.0-SNAPSHOT.jar:8.8.0-SNAPSHOT]
	at io.camunda.exporter.tasks.archiver.BatchOperationArchiverJob.archiveBatch(BatchOperationArchiverJob.java:50) ~[camunda-exporter-8.8.0-SNAPSHOT.jar:8.8.0-SNAPSHOT]
```

### Expected behavior

Archiver runs successfully without exception.

### Environment

SaaS

### Version

- Component Version: 8.8.0-SNAPSHOT


### Rootcause

[here](https://github.com/camunda/camunda/blob/dc1f45d55e255253898385eb4d503a4798ee250e/zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/archiver/BatchOperationArchiverJob.java#L64) the wrong value is passed as `idFieldName`.

### Solution ideas

_No response_

### Dev -> QA handover

- Resources:
- Versions to validate:
- Release version (in which version this feature will be released):


### Links

Reported in Slack: https://camunda.slack.com/archives/C06F0GLJNFM/p1744278329003539

Happened [here](Known issue https://github.com/camunda/camunda/issues/30877) as well.

```
 java.util.concurrent.CompletionException: co.elastic.clients.util.MissingRequiredPropertyException: Missing required property 'TermsQuery.field'
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture$Completion.run(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[?:?]
	at java.base/java.lang.Thread.run(Unknown Source) [?:?]
Caused by: co.elastic.clients.util.MissingRequiredPropertyException: Missing required property 'TermsQuery.field'
	at co.elastic.clients.util.ApiTypeHelper.requireNonNull(ApiTypeHelper.java:76) ~[elasticsearch-java-8.16.4.jar:?]
	at co.elastic.clients.elasticsearch._types.query_dsl.TermsQuery.<init>(TermsQuery.java:78) ~[elasticsearch-java-8.16.4.jar:?]
	at co.elastic.clients.elasticsearch._types.query_dsl.TermsQuery.<init>(TermsQuery.java:64) ~[elasticsearch-java-8.16.4.jar:?]
	at co.elastic.clients.elasticsearch._types.query_dsl.TermsQuery$Builder.build(TermsQuery.java:189) ~[elasticsearch-java-8.16.4.jar:?]
	at io.camunda.exporter.tasks.archiver.ElasticsearchArchiverRepository.buildIdTermsQuery(ElasticsearchArchiverRepository.java:248) ~[camunda-exporter-8.8.0-SNAPSHOT.jar:8.8.0-SNAPSHOT]
	at io.camunda.exporter.tasks.archiver.ElasticsearchArchiverRepository.lambda$reindexDocuments$9(ElasticsearchArchiverRepository.java:159) ~[camunda-exporter-8.8.0-SNAPSHOT.jar:8.8.0-SNAPSHOT]
	at co.elastic.clients.elasticsearch.core.reindex.Source$Builder.query(Source.java:337) ~[elasticsearch-java-8.16.4.jar:?]
	at io.camunda.exporter.tasks.archiver.ElasticsearchArchiverRepository.reindexDocuments(ElasticsearchArchiverRepository.java:159) ~[camunda-exporter-8.8.0-SNAPSHOT.jar:8.8.0-SNAPSHOT]
	at io.camunda.exporter.tasks.archiver.ArchiverRepository.moveDocuments(ArchiverRepository.java:41) ~[camunda-exporter-8.8.0-SNAPSHOT.jar:8.8.0-SNAPSHOT]
	at io.camunda.exporter.tasks.archiver.BatchOperationArchiverJob.moveBatch(BatchOperationArchiverJob.java:61) ~[camunda-exporter-8.8.0-SNAPSHOT.jar:8.8.0-SNAPSHOT]
	at io.camunda.exporter.tasks.archiver.BatchOperationArchiverJob.archiveBatch(BatchOperationArchiverJob.java:50) ~[camunda-exporter-8.8.0-SNAPSHOT.jar:8.8.0-SNAPSHOT]
	... 8 more 
```

--- a/zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/archiver/BatchOperationArchiverJob.java
+++ b/zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/archiver/BatchOperationArchiverJob.java
@@ -70,16 +70,15 @@ public class BatchOperationArchiverJob implements ArchiverJob {
     return CompletableFuture.completedFuture(0);
   }
 
-  private CompletableFuture<Integer> moveBatch(
-      final String finishDate, final List<String> processInstanceKeys) {
+  private CompletableFuture<Integer> moveBatch(final String finishDate, final List<String> ids) {
     return repository
         .moveDocuments(
             batchOperationTemplate.getFullQualifiedName(),
             batchOperationTemplate.getFullQualifiedName() + finishDate,
-            finishDate,
-            processInstanceKeys,
+            BatchOperationTemplate.ID,
+            ids,
             executor)
-        .thenApplyAsync(ok -> processInstanceKeys.size(), executor);
+        .thenApplyAsync(ok -> ids.size(), executor);
   }
 
   @Override

--- a/zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/tasks/archiver/BatchOperationArchiverJobTest.java
+++ b/zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/tasks/archiver/BatchOperationArchiverJobTest.java
@@ -68,7 +68,7 @@ final class BatchOperationArchiverJobTest {
             new DocumentMove(
                 batchOperationTemplate.getFullQualifiedName(),
                 batchOperationTemplate.getFullQualifiedName() + "2024-01-01",
-                repository.batch.finishDate(),
+                BatchOperationTemplate.ID,
                 List.of("1", "2", "3"),
                 executor));
   }

